tests/src/core/testqgslayerdefinition.cpp

```diff
@@ -21,6 +21,7 @@
 
 #include <qgsapplication.h>
 #include <qgsproject.h>
+#include <qgsvectorlayer.h>
```

Comment:
Just a heads up -- this file is in the skip list for Travis, so your new test is basically a noop...

Reply:
Right, I moved the test.